sw/device/lib/dif/autogen/dif_keymgr_autogen_unittest.cc

```diff
@@ -0,0 +1,295 @@
+
+  EXPECT_EQ(dif_keymgr_irq_get_state(nullptr, &irq_snapshot), kDifBadArg);
+
+  EXPECT_EQ(dif_keymgr_irq_get_state(&keymgr_, nullptr), kDifBadArg);
+
+  EXPECT_EQ(dif_keymgr_irq_get_state(nullptr, nullptr), kDifBadArg);
+}
+
+TEST_F(IrqGetStateTest, SuccessAllRaised) {
+  dif_keymgr_irq_state_snapshot_t irq_snapshot = 0;
+
+  EXPECT_READ32(KEYMGR_INTR_STATE_REG_OFFSET,
+                std::numeric_limits<uint32_t>::max());
+  EXPECT_EQ(dif_keymgr_irq_get_state(&keymgr_, &irq_snapshot), kDifOk);
+  EXPECT_EQ(irq_snapshot, std::numeric_limits<uint32_t>::max());
+}
+
+TEST_F(IrqGetStateTest, SuccessNoneRaised) {
+  dif_keymgr_irq_state_snapshot_t irq_snapshot = 0;
+
+  EXPECT_READ32(KEYMGR_INTR_STATE_REG_OFFSET, 0);
+  EXPECT_EQ(dif_keymgr_irq_get_state(&keymgr_, &irq_snapshot), kDifOk);
+  EXPECT_EQ(irq_snapshot, 0);
+}
+
+class IrqIsPendingTest : public KeymgrTest {};
+
+TEST_F(IrqIsPendingTest, NullArgs) {
+  bool is_pending;
+
+  EXPECT_EQ(
+      dif_keymgr_irq_is_pending(nullptr, kDifKeymgrIrqOpDone, &is_pending),
+      kDifBadArg);
+
+  EXPECT_EQ(dif_keymgr_irq_is_pending(&keymgr_, kDifKeymgrIrqOpDone, nullptr),
+            kDifBadArg);
+
+  EXPECT_EQ(dif_keymgr_irq_is_pending(nullptr, kDifKeymgrIrqOpDone, nullptr),
+            kDifBadArg);
+}
+
+TEST_F(IrqIsPendingTest, BadIrq) {
+  bool is_pending;
+  // All interrupt CSRs are 32 bit so interrupt 32 will be invalid.
+  EXPECT_EQ(
+      dif_keymgr_irq_is_pending(&keymgr_, (dif_keymgr_irq_t)32, &is_pending),
+      kDifBadArg);
+}
+
+TEST_F(IrqIsPendingTest, Success) {
+  bool irq_state;
+
+  // Get the first IRQ state.
+  irq_state = false;
+  EXPECT_READ32(KEYMGR_INTR_STATE_REG_OFFSET,
+                {{KEYMGR_INTR_STATE_OP_DONE_BIT, true}});
+  EXPECT_EQ(
+      dif_keymgr_irq_is_pending(&keymgr_, kDifKeymgrIrqOpDone, &irq_state),
+      kDifOk);
+  EXPECT_TRUE(irq_state);
+
+  // Get the last IRQ state.
+  irq_state = true;
+  EXPECT_READ32(KEYMGR_INTR_STATE_REG_OFFSET,
+                {{KEYMGR_INTR_STATE_OP_DONE_BIT, false}});
+  EXPECT_EQ(
+      dif_keymgr_irq_is_pending(&keymgr_, kDifKeymgrIrqOpDone, &irq_state),
+      kDifOk);
+  EXPECT_FALSE(irq_state);
+}
+
+class IrqAcknowledgeTest : public KeymgrTest {};
+
+TEST_F(IrqAcknowledgeTest, NullArgs) {
+  EXPECT_EQ(dif_keymgr_irq_acknowledge(nullptr, kDifKeymgrIrqOpDone),
+            kDifBadArg);
+}
+
+TEST_F(IrqAcknowledgeTest, BadIrq) {
+  EXPECT_EQ(dif_keymgr_irq_acknowledge(nullptr, (dif_keymgr_irq_t)32),
+            kDifBadArg);
+}
+
+TEST_F(IrqAcknowledgeTest, Success) {
+  // Clear the first IRQ state.
+  EXPECT_WRITE32(KEYMGR_INTR_STATE_REG_OFFSET,
+                 {{KEYMGR_INTR_STATE_OP_DONE_BIT, true}});
+  EXPECT_EQ(dif_keymgr_irq_acknowledge(&keymgr_, kDifKeymgrIrqOpDone), kDifOk);
+
+  // Clear the last IRQ state.
+  EXPECT_WRITE32(KEYMGR_INTR_STATE_REG_OFFSET,
+                 {{KEYMGR_INTR_STATE_OP_DONE_BIT, true}});
+  EXPECT_EQ(dif_keymgr_irq_acknowledge(&keymgr_, kDifKeymgrIrqOpDone), kDifOk);
```

Comment:
nit: should we special case these tests for devices with a single irq so that we don't test the same thing twice?